src/components/HomeView.jsx

```diff
@@ -1,15 +1,67 @@
-import React from "react";
+import React, { useState } from "react";
 import { useNavigate } from "react-router-dom";
+import { useSelector, useDispatch } from 'react-redux';
+import Authentication from "../modules/Authentication";
 
 const HomeView = () => {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { errorMessage } = useSelector(state => state);
+  const [userData, setUserData] = useState({});
+
+  const handleChange = (event) => {
+    setUserData({
+      ...userData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = async () => {
+    const response = await Authentication.signIn(
+      userData.email,
+      userData.password
+    );
+    if(response.status === 'success'){
+      dispatch({ type: 'SET_WELCOME_MESSAGE', payload: `Welcome ${response.data.name}!` });
+      dispatch({ type: 'SET_CURRENT_USER', payload: response.data });
```

Comment:
You are storing the user's name twice: once in `SET_WELCOME_MESSAGE` and once in `SET_CURRENT_USER`. I suggest removing the `SET_WELCOME_MESSAGE` reducer from application state and instead grabbing the user's name from `currentUser` for your dashboard.